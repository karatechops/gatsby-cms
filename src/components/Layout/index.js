import React from 'react';
import PropTypes from 'prop-types';
import { grommet, Anchor, Grommet, Box, Text } from 'grommet';
import { deepMerge } from 'grommet/utils';
import { css } from 'styled-components';

import './reset.css';
import Content from '../Content';

const theme = deepMerge(
  {
    heading: {
      extend: css`
        font-weight: 100;
      `,
    },
    paragraph: {
      extend: css`
        font-weight: 100;
        max-width: 100%;
      `,
    },
  },
  grommet,
);

class Layout extends React.Component {
  render() {
    const { children } = this.props;

    return (
      <Grommet
        full
        theme={theme}
        style={{
          display: 'flex',
          flexDirection: 'column',
          maxWidth: '100%',
          minHeight: '100vh',
          height: 'auto',
        }}
      >
        <Box flex as="main">
          {children}
        </Box>
        <Box as="footer" background="dark-1" pad={{ vertical: 'medium' }}>
          <Content direction="row" justify="center">
            <Text>
              Â© {new Date().getFullYear()}, Built with{' '}
              <Anchor href="https://grommet.io">
                <span role="img" aria-label="love">
                  ðŸ’œ
                </span>
              </Anchor>
            </Text>
          </Content>
        </Box>
      </Grommet>
    );
  }
}

Layout.propTypes = {
  children: PropTypes.node.isRequired,
};

export default Layout;
