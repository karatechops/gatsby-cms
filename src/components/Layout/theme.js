import { grommet } from 'grommet';
import { deepMerge } from 'grommet/utils';
import { css } from 'styled-components';

const theme = deepMerge(
  {
    heading: {
      extend: css`
        font-weight: 100;
      `,
    },
    paragraph: {
      extend: css`
        font-weight: 100;
        max-width: 100%;
      `,
    },
  },
  grommet,
);

export default theme;
