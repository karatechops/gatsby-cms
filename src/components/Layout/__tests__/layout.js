import React from 'react';
import renderer from 'react-test-renderer';

import Layout from '../index';

describe('Layout', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(<Layout width="medium">This is my app wrapper</Layout>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
