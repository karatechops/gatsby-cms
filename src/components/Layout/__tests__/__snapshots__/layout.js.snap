// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout renders correctly 1`] = `
<div
  className="StyledGrommet-sc-19lkkz7-0 cqQlmF"
  style={
    Object {
      "display": "flex",
      "flexDirection": "column",
      "height": "auto",
      "maxWidth": "100%",
      "minHeight": "100vh",
    }
  }
>
  <div
    className="StyledBox-sc-13pk1d4-0 dOfRrV"
  >
    <a
      href="/"
    >
      <span
        className="StyledText-sc-1sadyjn-0 dPfkAU"
      >
        My site
      </span>
    </a>
    <div
      className="StyledBox__StyledBoxGap-sc-13pk1d4-1 iChEkS"
    />
    <a
      href="/blog"
    >
      <span
        className="StyledText-sc-1sadyjn-0 dPfkAU"
      >
        Blog
      </span>
    </a>
  </div>
  <main
    className="StyledBox-sc-13pk1d4-0 gYrpnw"
  >
    This is my app wrapper
  </main>
  <footer
    className="StyledBox-sc-13pk1d4-0 cJOxqP"
  >
    <div
      className="StyledBox-sc-13pk1d4-0 ldwCNw"
    >
      <span
        className="StyledText-sc-1sadyjn-0 fWSbXS"
      >
        Â© 
        2019
        , Built with
         
        <a
          className="StyledAnchor-sc-1rp7lwl-0 llbwuF"
          href="https://grommet.io"
          onBlur={[Function]}
          onFocus={[Function]}
        >
          <span
            aria-label="love"
            role="img"
          >
            ðŸ’œ
          </span>
        </a>
      </span>
    </div>
  </footer>
</div>
`;
